### PR TITLE
[ACA-20] Favorite Libraries

### PR DESCRIPTION
**Pull request by pionnegru**
## PR Checklist
Please check if your PR fulfills the following requirements:

```
- [ ] The commit message follows our guidelines: https://github.com/Alfresco/alfresco-content-app/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
```

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application / Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/app/components/favorite-libraries/favorite-libraries.component.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+              <aca-toolbar-action [actionRef]="entry"></aca-toolbar-action>
+          </ng-container>
+      </adf-toolbar>
+  </app-page-layout-header>
+
+  <app-page-layout-content>
+      <div class="main-content">
+          <adf-document-list #documentList
+              acaDocumentList
+              custom="-bla-"
```
Comment (DenysVuika) [inline src/app/components/favorite-libraries/favorite-libraries.component.html]: Might not be relevant to the document list

File: src/app/extensions/extension.service.ts [hunk 2]
```diff
@@ -75,6 +75,7 @@ export class AppExtensionService implements RuleContext {
   documentListPresets: {
     files: Array<DocumentListPresetRef>;
     libraries: Array<DocumentListPresetRef>;
+    favorite_libraries: Array<DocumentListPresetRef>;
```
Comment (DenysVuika) [inline src/app/extensions/extension.service.ts]: can we name it `favoriteLibraries` instead?

File: src/app/extensions/extension.service.ts [hunk 3]
```diff
@@ -162,6 +164,10 @@ export class AppExtensionService implements RuleContext {
     this.documentListPresets = {
       files: this.getDocumentListPreset(config, 'files'),
       libraries: this.getDocumentListPreset(config, 'libraries'),
+      favorite_libraries: this.getDocumentListPreset(
```
Comment (DenysVuika) [inline src/app/extensions/extension.service.ts]:
```suggestion
      favoriteLibraries: this.getDocumentListPreset(
```

File: src/app/extensions/extension.service.ts [hunk 4]
```diff
@@ -162,6 +164,10 @@ export class AppExtensionService implements RuleContext {
...
+        config,
+        'favorite_libraries'
```
Comment (DenysVuika) [inline src/app/extensions/extension.service.ts]:
```suggestion
        'favoriteLibraries'
```

File: src/assets/app.extensions.json [hunk 5]
```diff
@@ -979,6 +1003,44 @@
           "desktopOnly": true
         }
       ],
+      "favorite_libraries": [
```
Comment (DenysVuika) [inline src/assets/app.extensions.json]:
```suggestion
      "favoriteLibraries": [
```

File: src/app/extensions/evaluators/navigation.evaluators.ts [hunk 6]
```diff
@@ -75,7 +75,7 @@ export function isLibraries(
   ...args: RuleParameter[]
 ): boolean {
   const { url } = context.navigation;
-  return url && url.endsWith('/libraries');
```
Comment (DenysVuika) [inline src/app/extensions/evaluators/navigation.evaluators.ts]: How that behave when drilling down the files in the Library? previously this one was checking just the root one (ends with) but not the folders inside

File: src/assets/i18n/en.json [hunk 7]
```diff
@@ -160,7 +183,8 @@
                 "LOCATION": "Location",
                 "SHARED_BY": "Shared by",
                 "DELETED_ON": "Deleted",
-                "DELETED_BY": "Deleted by"
+                "DELETED_BY": "Deleted by",
+                "ROLE": "My Role"
```
Comment (adinapitul) [inline src/assets/i18n/en.json]: ROLE is defined twice - see line 176

File: e2e/suites/application/page-titles.test.ts [hunk 8]
```diff
@@ -81,8 +81,10 @@ describe('Page titles', () => {
     });
 
     it('File Libraries page - [C217158]', async () => {
+      const parent = SIDEBAR_LABELS.FILE_LIBRARIES;
       const label = SIDEBAR_LABELS.FILE_LIBRARIES;
```
Comment (adinapitul) [inline e2e/suites/application/page-titles.test.ts]: should "label" be in fact SIDEBAR_LABELS.MY_LIBRARIES ?

Comment (adinapitul) [reply e2e/suites/application/page-titles.test.ts]: do we have different page titles for My Libraries and Favorite Libraries ?

Comment (pionnegru) [reply e2e/suites/application/page-titles.test.ts]: yes

File: src/app/extensions/evaluators/navigation.evaluators.ts (hunk 6 above)
Comment (pionnegru) [reply src/app/extensions/evaluators/navigation.evaluators.ts]: was a bug in my previous commit.

**Pull request merged**